HW_5/index.js

```diff
@@ -0,0 +1,159 @@
+    li.appendChild(deleteElement);
+    deleteElement.addEventListener('click', clearSingleTask);
+
+    taskList.appendChild(li);
+
+}
+
+function editOneElement(edit, textElement, editButton) {
+    const editTask = document.createElement('input');
+    textElement.remove();
+    editButton.remove();
+    editTask.setAttribute('type', 'text');
+    editTask.value = textElement.textContent;
+    editTask.addEventListener("keypress", function (event) {
+        if (event.key === "Enter") {
+            event.preventDefault();
+            editTask.remove();
+            console.log(edit.textContent)
+            textElement.textContent = editTask.value;
+            edit.insertBefore(editButton, edit.children[0]);
+            edit.insertBefore(textElement, edit.children[0]);
+
+            const nodes = Array.prototype.slice.call(taskList.children);
+            const index = nodes.indexOf(textElement.parentElement);
+            updateTaskInLocalStorage(textElement.textContent, index);
+        }
+    })
+
+    edit.insertBefore(editTask, edit.children[0]);
+    console.log(edit)
+
+}
+
+
+function renderTasks() {
+    let tasks;
+
+    if (localStorage.getItem('tasks') !== null) {
+        tasks = JSON.parse(localStorage.getItem('tasks'));
+    } else {
+        tasks = [];
+    }
+
+    tasks.forEach((task) => {
+        createSingleTaskElement(task);
+    })
+}
+
+function createTask(event) {
+    event.preventDefault();
+    if (taskInput.value.trim() === '') {
+        return;
+    }
+
+    createSingleTaskElement(taskInput.value);
+
+    storeTaskInLocalStorage(taskInput.value);
+
+    taskInput.value = '';
+}
+
+function updateTaskInLocalStorage(updatedTask, index) {
+    const tasks = getTasksFromLocalStorage();
+    tasks[index] = updatedTask;
+    setTasksToLocalStorage(tasks);
+}
+
+function storeTaskInLocalStorage(task) {
+    let tasks;
+    if (localStorage.getItem('tasks') !== null) {
+        tasks = JSON.parse(localStorage.getItem('tasks'));
+    } else {
+        tasks = [];
+    }
+    tasks.push(task);
+    localStorage.setItem('tasks', JSON.stringify(tasks));
+}
+
+function clearAllTasks() {
+    if (confirm('Are you sure?')) {
+        localStorage.clear();
+        taskList.innerHTML = '';
+    }
+}
+
+
+
+function clearSingleTask(event) {
+    let iconContainer = event.target.parentElement;
+
+    const nodes = Array.prototype.slice.call(taskList.children);
+    const index = nodes.indexOf(iconContainer.parentElement);
+    if (iconContainer.classList.contains('delete-item')) {
+        if (confirm('Are you sure?')) {
+            iconContainer.parentElement.remove();
+            removeTaskFromLocalStorage(iconContainer.parentElement, index);
+        }
+
+    }
+}
+
+function removeTaskFromLocalStorage(tasktoRemove, removeIndex) {
+    let tasks;
+
+    if (localStorage.getItem('tasks') !== null) {
+        tasks = JSON.parse(localStorage.getItem('tasks'));
+    } else {
+        tasks = [];
+    }
```

Comment:
```suggestion
let tasks = getTasksFromLocalStorage();
```

Функція getTasksFromLocalStorage ма/
є в собі цей же функціонал